[quant][fix][ez] Fix quant_type classification for fp16, fp16

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #48107 [quant][refactor] Move some util functions from torch/quantization/fx/utils.py to torch/quantization/utils.py
* **#48073 [quant][fix][ez] Fix quant_type classification for fp16, fp16**
* #48069 [quant][fix] Fix quant type classification for float_qparam qconfig
* #48067 [quant][graphmode][fx] Keep linear op unchanged when qconfig is not supported

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25011799](https://our.internmc.facebook.com/intern/diff/D25011799)

jerryzh168 committed:
[quant][fix][ez] Fix quant_type classification for fp16, fp16

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fix][ez] Fix quant_type classification for fp16, fp16"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25011799](https://our.internmc.facebook.com/intern/diff/D25011799)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fix][ez] Fix quant_type classification for fp16, fp16"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25011799](https://our.internmc.facebook.com/intern/diff/D25011799)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fix][ez] Fix quant_type classification for fp16, fp16"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25011799](https://our.internmc.facebook.com/intern/diff/D25011799)

[ghstack-poisoned]

supriyar reviewed (approved):
Failing tests seem relevant to previous PR.

jerryzh168 committed:
Update on "[quant][fix][ez] Fix quant_type classification for fp16, fp16"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25011799](https://our.internmc.facebook.com/intern/diff/D25011799)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fix][ez] Fix quant_type classification for fp16, fp16"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25011799](https://our.internmc.facebook.com/intern/diff/D25011799)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fix][ez] Fix quant_type classification for fp16, fp16"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25011799](https://our.internmc.facebook.com/intern/diff/D25011799)

[ghstack-poisoned]
